checker/src/main/java/org/checkerframework/checker/signedness/jdk.astub

```diff
@@ -94,6 +98,11 @@ class Long {
     public static @Unsigned long remainderUnsigned(@Unsigned long dividend, @Unsigned long divisor);
 }
 
+class Object {
+    protected @Unsigned Object clone() throws CloneNotSupportedException;
```

Comment:
`@Unsigned` should not be applied to non-numeric types.

Reply:
Array is an object in java and if we want to clone an integer or byte array that is unsigned then due to this change we will not get error?
for ex.-
import java.util.Arrays; 
import org.checkerframework.checker.signedness.qual.*;
class ex { 

	private static @Unsigned byte[] de(@Unsigned byte[] ci){

		@Unsigned byte b[]=ci.clone();
		return b;
	}
}
This will give error if this change is not made.

Reply:
I agree the change to the annotated JDK will fix this use case.
However, it will break others:
 * cloning a signed byte array will result in an unsigned byte array.  This introduces unsoundness.
 * cloning a string will result in an unsigned string.  This introduces illegal annotations.

It is not acceptable to fix one problem while introducing both unsoundness and illegal annotations.

Reply:
Can you think of a different way to achieve your goals?

Reply:
Yes, If we use @PolySigned then it should work fine.

Reply:
Yes, that is a better choice.  Please write a few test cases to ensure that the annotations behave as you expect.  Thanks!

Reply:
Yes, I checked for ex-
import java.lang.*; 
import org.checkerframework.checker.signedness.qual.*;
class ex2 { 

	private static @Unsigned byte[] de(byte[] ci){

		@Unsigned byte b[]=ci.clone();
		return b;
	}
}
This is giving error which would not occur earlier due to @Unsigned, but now due to @PolySigned it is giving the error. Rest all are working fine.

Reply:
Thanks for checking that.  It is usually a good idea to have both positive and negative test cases (that is, both code that typechecks and code that produces a type-checking error), and to try with multiple types (including non-arrays which was the original concern).

It may be that your changes are correct, but it's good to double-check it.
Thanks!

Reply:
Sorry, but I am unable to understand "non arrays" such as? Can you give examples.
Thanks!

Reply:
You could call clone on a String, or an Integer, or other types that are not arrays.

Reply:
In order to expose the clone() method to clients, we have to declare it in our subclass with a more liberal access modifier, either public or package private. However, we cannot extend java.lang.Integer because it is a final class. Also note that java.lang.Integer does not implement Cloneable then how can we use clone() method with Integer or string etc?

Reply:
Yes, you have identified a serious and interesting limitation.  We are not sure how to deal with it.
I'm not sure it is a problem in your case, though.  Why do you need to change the contract of clone?  Isn't the contract on the Object class sufficient for all subclasses?

Reply:
I am not able to apply clone in Integer or String.

Reply:
Why not?  Can you show test code and the results?

Reply:
If we simply run in java-
import java.lang.*; 
class ex2 { 

	private static Integer de(Integer ci){

		Integer b=ci.clone();
		return b;
	}
}
then it will give error.-
gsoc/ex2.java:6: error: clone() has protected access in Object
		Integer b=ci.clone();
		            ^
gsoc/ex2.java:6: error: incompatible types: Object cannot be converted to Integer
		Integer b=ci.clone();
		                  ^
2 errors

Reply:
2nd error can be corectted by-
 Integer b=(Integer)ci.clone();
but how the 1st one.